src/core/linux-aio.cc

```diff
@@ -170,8 +171,12 @@ int io_pgetevents(aio_context_t io_context, long min_nr, long nr, io_event* even
 void setup_aio_context(size_t nr, linux_abi::aio_context_t* io_context) {
     auto r = io_setup(nr, io_context);
     if (r < 0) {
-        char buf[1024];
-        char *msg = strerror_r(errno, buf, sizeof(buf));
+	char buf[1024];
+#ifdef __GLIBC__
+        const char *msg = strerror_r(errno, buf, 0);
```

Comment:
Why did you stop using sizeof(buf) and changed it to 0?
I think that if errno has some unrecognized number, the buffer *will* be used (to contain a string like "unknown error 1234") so better safe than sorry, do pass the correct sizeof(buf) and not 0.
(as an aside, if you only needed 1024 bytes for the Posix variant, please move it to the else).

Reply:
> Why did you stop using sizeof(buf) and changed it to 0? I think that if errno has some unrecognized number, the buffer _will_ be used (to contain a string like "unknown error 1234") so better safe than sorry,
> do pass the correct sizeof(buf) and not 0. (as an aside, if you only needed 1024 bytes for the Posix variant, please move it to the else).

i think it was my idea. after a second thought and cross-checking the document by reading the [glibc code](https://github.com/bminor/glibc/blob/6333a6014f32c7ced36ced610d5a5cecbafba6c3/string/_strerror.c#L24-L35), i agree with you. we should respect glibc's manpage.

> his may be either a pointer to a string that the function stores in buf, or a pointer to some (immutable) static string  (in which case buf is unused).  If the function stores a string in buf, then at most buflen bytes are stored (the string may be truncated if buflen is too small and errnum is unknown).  The string always  includes  a  terminating  null         byte ('\0').

Reply:
note to myself, 128 bytes would be more than enough. it's not that important though.

Reply:
okay will change